### PR TITLE
Add normalize method and fix java bug

### PR DESCRIPTION
**Pull request by fergmastaflex**
The normalize method returns a number between 0 and 1 based on the results of the existing distance method. There are methods for Java, C and pure Ruby.

In order to implement this, we had to fix a bug in the Java code that caused the `for` loops to skip the final character in the strings.

@abrandoned @liveh2o @mmmries @ShaneSmit @mattnichols

### CODE REVIEW

File: ext/levenshtein/levenshtein.c [hunk 1]
```diff
@@ -104,3 +104,15 @@ int levenshtein_extern(char* a, char* b, int max_distance)
 
   return distance;
 }
+
+float normalized_levenshtein_extern(char* a, char* b, int max_distance)
```
Comment (abrandoned) [inline ext/levenshtein/levenshtein.c:108]: don't you want the normalized distance to be 1 if it exceeds the max? (not 0)

File: ext/levenshtein/levenshtein.c [hunk 2]
```diff
@@ -104,3 +104,15 @@ int levenshtein_extern(char* a, char* b, int max_distance)
...
+{
+  int a_len = (a == NULL) ? 0 : strlen(a);
+  int b_len = (b == NULL) ? 0 : strlen(b);
+
+  int max_string_length = (a_len > b_len) ? a_len : b_len;
+  if(max_string_length == 0) {
+    return 0;
```
Comment (fergmastaflex) [inline ext/levenshtein/levenshtein.c:115]: @abrandoned This return is if both strings are blank.

This line will return the length of the longest string to the method I created, which will result in dividing by itself and returning 1: https://github.com/mxenabled/vladlev/blob/ferg/normalize_and_bug_fix/ext/levenshtein/levenshtein.c#L75

File: spec/lib/levenshtein_spec.rb [hunk 3]
```diff
@@ -48,6 +49,15 @@
     specify{ Vladlev.distance('*&^%$+_=-)(*&^%$#@!~123456789', '*&^%$+_=-)(*&^%$#@!~').should equal(9) }
     specify{ Vladlev.distance('*&^%$+_=-)(*&^%$#@!~', "").should equal(20) }
   end
+
+  describe "normalized distance" do
+    specify{ expect(Vladlev.distance("cur", "cut", 10)).to eq(1) }
+    specify{ expect(Vladlev.get_normalized_distance("hello", "hello")).to eq(0.0) }
```
Comment (ShaneSmit) [inline spec/lib/levenshtein_spec.rb:56]: Should switch this unit test call from `distance()` to `get_normalized_distance()`.

Comment (fergmastaflex) [inline spec/lib/levenshtein_spec.rb:56]: agreed :+1:

File: lib/vladlev/levenshtein.rb [hunk 4]
```diff
@@ -39,5 +39,11 @@ def self.distance(str1, str2, maximum_allowable_distance = 9999)
 
       return broke_max ? shortest_string.size : calculation_grid[longest_string.size - 1]
     end
+
+    def self.normalized_distance(str1, str2, maximum_allowable_distance = 9999)
+      longest_string_length = (str1 > str2) ? str1.length : str2.length
+      return 0 if longest_string_length
```
Comment (ShaneSmit) [inline lib/vladlev/levenshtein.rb]: Shouldn't this be

``` ruby
return 0 if longest_string_length == 0
```

**Comment by ShaneSmit** [comment]
:shipit:

**Pull request merged**